Controller/Methods/Delete.php

```diff
@@ -0,0 +1,128 @@
+<?php
+namespace Omise\Payment\Controller\Methods;
+
+use Magento\Framework\App\Request\Http;
+use Magento\Framework\Exception\NotFoundException;
+
+class Delete extends \Magento\Framework\App\Action\Action
+{
+    const WRONG_REQUEST = 1;
+
+    const WRONG_TOKEN = 2;
+
+    const ACTION_EXCEPTION = 3;
+
+    /**
+     * @var array
+     */
+    private $errorsMap = [];
+
+    /**
+     * @var Magento\Customer\Model\Session
+     */
+    private $customerSession;
+
+    /**
+     * @var Omise\Payment\Model\Customer
+     */
+    private $customer;
+
+    /**
+     * @param Context  $context
+     * @param Session  $customerSession
+     * @param Customer $customer
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
+        \Magento\Customer\Model\Session       $customerSession,
+        \Omise\Payment\Model\Customer         $customer
+    ) {
+        parent::__construct($context, $customerSession);
+        $this->customerSession = $customerSession;
+        $this->customer        = $customer;
+
+        $this->errorsMap = [
+            self::WRONG_TOKEN      => __('No token found.'),
+            self::WRONG_REQUEST    => __('Wrong request.'),
+            self::ACTION_EXCEPTION => __('Deletion failure. Please try again.'),
+        ];
+    }
+
+    /**
+     * Dispatch request
+     *
+     * @return Magento\Framework\Controller\ResultInterface|Magento\Framework\App\ResponseInterface
+     * @throws NotFoundException
+     */
+    public function execute()
+    {
+        $request = $this->_request;
+        if (!$request instanceof Http) {
+            return $this->createErrorResponse(self::WRONG_REQUEST);
+        }
+
+        $cardId = $this->getCardID($request);
+
+        if ($cardId === null) {
+            return $this->createErrorResponse(self::WRONG_TOKEN);
+        }
+
+        try {
+            $this->customer->deleteCard($cardId);
+        } catch (\Exception $e) {
+            return $this->createErrorResponse(self::ACTION_EXCEPTION);
+        }
+
+        return $this->createSuccessMessage();
+    }
+
+    /**
+     * @param int $errorCode
+     * @return Magento\Framework\App\ResponseInterface
+     */
+    private function createErrorResponse($errorCode)
+    {
+        $this->messageManager->addErrorMessage(
+            $this->errorsMap[$errorCode]
+        );
+
+        return $this->_redirect('omise/methods/cards');
+    }
+
+    /**
+     * @return Magento\Framework\App\ResponseInterface
+     */
+    private function createSuccessMessage()
+    {
+        $this->messageManager->addSuccessMessage(
+            __('Stored Payment Method was successfully removed')
```

Comment:
Should this refer to card?

Reply:
it does not matter.. does it? it is just confirmation, fully describing what just happened.

Reply:
Well maybe, but if there are only cards, the wording should reflect that. Also, the capitalisation needs fixing

Reply:
A well designed UX should always remove all unnecessary questions from a user's mind

Reply:
@guzzilar Thoughts?

Reply:
Just wonder why "delete a saved card" should return `Stored Payment Method was successfully removed`, not "The card was successfully removed".

Reply:
credit / depit card is payment method.

Reply:
But you don't delete payment method, you delete card, right?

Reply:
'Card' is definitely much clearer here